core/scripts/chaincli/handler/scrape_node_config.go

```diff
@@ -0,0 +1,283 @@
+	DisplayName           string         `json:"displayName"`
+	NodeAddress           []string       `json:"nodeAddress"`
+	Ocr2ConfigPublicKey   []string       `json:"ocr2ConfigPublicKey"`
+	Ocr2Id                []string       `json:"ocr2Id"`
+	Ocr2OffchainPublicKey []string       `json:"ocr2OffchainPublicKey"`
+	Ocr2OnchainPublicKey  []string       `json:"ocr2OnchainPublicKey"`
+	OcrSigningAddress     []string       `json:"ocrSigningAddress"`
+	PayeeAddress          common.Address `json:"payeeAddress"`
+	PeerId                []string       `json:"peerId"`
+	Status                string         `json:"status"`
+}
+
+func (node NodeInfo) Equals(ni NodeInfo, log logger.Logger) bool {
+	diffs := 0
+
+	if len(node.CSAKeys) != len(ni.CSAKeys) {
+		log.Errorf("CSA Keys length differs. The node returns %d but weiwatcher has %d", len(node.CSAKeys), len(ni.CSAKeys))
+	}
+	for i, ci := range node.CSAKeys {
+		if !ci.Equals(ni.CSAKeys[i]) {
+			diffs++
+			log.Errorf("CSA Info differs. The node returns %s but weiwatcher has %s", ci, ni.CSAKeys[i])
+		}
+	}
+
+	if !cmp.Equal(node.Ocr2Id, ni.Ocr2Id) {
+		diffs++
+		log.Errorf("OCR2 ID differs. The node returns %s but weiwatcher has %s", node.Ocr2Id, ni.Ocr2Id)
+	}
+
+	if !cmp.Equal(node.NodeAddress, ni.NodeAddress) {
+		diffs++
+		log.Errorf("Node address differs. The node returns %s but weiwatcher has %s", node.NodeAddress, ni.NodeAddress)
+	}
+
+	if !cmp.Equal(node.PeerId, ni.PeerId) {
+		diffs++
+		log.Errorf("Peer Id differs. The node returns %s but weiwatcher has %s", node.PeerId, ni.PeerId)
+	}
+
+	if !cmp.Equal(node.Ocr2OffchainPublicKey, ni.Ocr2OffchainPublicKey) {
+		diffs++
+		log.Errorf("OCR2 Offchain Public Key differs. The node returns %s but weiwatcher has %s", node.Ocr2OffchainPublicKey, ni.Ocr2OffchainPublicKey)
+	}
+
+	if !cmp.Equal(node.Ocr2OnchainPublicKey, ni.Ocr2OnchainPublicKey) {
+		diffs++
+		log.Errorf("OCR2 Onchain Public Key differs. The node returns %s but weiwatcher has %s", node.Ocr2OnchainPublicKey, ni.Ocr2OnchainPublicKey)
+	}
+
+	if !cmp.Equal(node.Ocr2ConfigPublicKey, ni.Ocr2ConfigPublicKey) {
+		diffs++
+		log.Errorf("OCR2 Config Public Key differs. The node returns %s but weiwatcher has %s", node.Ocr2ConfigPublicKey, ni.Ocr2ConfigPublicKey)
+	}
+
+	return diffs == 0
+}
+
+func (h *baseHandler) ScrapeNodes() {
+	log, closeLggr := logger.NewLogger()
+	logger.Sugared(log).ErrorIfFn(closeLggr, "Failed to close logger")
+
+	ctx := context.Background()
+	h.scrapeNodes(ctx, log)
+}
+
+func (h *baseHandler) scrapeNodes(ctx context.Context, log logger.Logger) {
+	log.Warn("This scrapes node address, peer ID, CSA node address, CSA public key, OCR2 ID, OCR2 config pub key, OCR2 onchain pub key, and OCR2 offchain pub key.")
+	log.Warn("This does NOT scrape for payee address, admin address etc. Please verify that manually.")
+	cls := make([]cmd.HTTPClient, len(h.cfg.KeeperURLs))
+	for i := range h.cfg.KeeperURLs {
+		url := h.cfg.KeeperURLs[i]
```

Comment:
No big deal, but you could get rid of this line if you changed the loop above to be:

```
for i, url := range h.cfg.KeeperURLs {
```